server/command_hooks.go

```diff
@@ -63,100 +81,243 @@ func (p *Plugin) emitStatusChange() {
+		errorMessage := "Failed to open Interactive Dialog"
+		p.API.LogError(errorMessage, "err", err.Error())
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         errorMessage,
+		}
+	}
+	return &model.CommandResponse{}
+}
+
+func getDialogWithSampleElements() model.Dialog {
+	return model.Dialog{
+		CallbackId: "somecallbackid",
+		Title:      "Test Title",
+		IconURL:    "http://www.mattermost.org/wp-content/uploads/2016/04/icon.png",
+		Elements: []model.DialogElement{{
+			DisplayName: "Display Name",
+			Name:        "realname",
+			Type:        "text",
+			Default:     "default text",
+			Placeholder: "placeholder",
+			HelpText:    "This a regular input in an interactive dialog triggered by a test integration.",
+		}, {
+			DisplayName: "Email",
+			Name:        dialogElementNameEmail,
+			Type:        "text",
+			SubType:     "email",
+			Placeholder: "placeholder@bladekick.com",
+			HelpText:    "This a regular email input in an interactive dialog triggered by a test integration.",
+		}, {
+			DisplayName: "Number",
+			Name:        dialogElementNameNumber,
+			Type:        "text",
+			SubType:     "number",
+		}, {
+			DisplayName: "Display Name Long Text Area",
+			Name:        "realnametextarea",
+			Type:        "textarea",
+			Placeholder: "placeholder",
+			Optional:    true,
+			MinLength:   5,
+			MaxLength:   100,
+		}, {
+			DisplayName: "User Selector",
+			Name:        "someuserselector",
+			Type:        "select",
+			Placeholder: "Select a user...",
+			HelpText:    "Choose a user from the list.",
+			Optional:    true,
+			MinLength:   5,
+			MaxLength:   100,
+			DataSource:  "users",
+		}, {
+			DisplayName: "Channel Selector",
+			Name:        "somechannelselector",
+			Type:        "select",
+			Placeholder: "Select a channel...",
+			HelpText:    "Choose a channel from the list.",
+			Optional:    true,
+			MinLength:   5,
+			MaxLength:   100,
+			DataSource:  "channels",
+		}, {
+			DisplayName: "Option Selector",
+			Name:        "someoptionselector",
+			Type:        "select",
+			Placeholder: "Select an option...",
+			HelpText:    "Choose an option from the list.",
+			Options: []*model.PostActionOptions{{
+				Text:  "Option1",
+				Value: "opt1",
+			}, {
+				Text:  "Option2",
+				Value: "opt2",
+			}, {
+				Text:  "Option3",
+				Value: "opt3",
+			}},
+		}},
+		SubmitLabel:    "Submit",
+		NotifyOnCancel: true,
+		State:          "somestate",
+	}
+}
+
+func getDialogWithoutElements() model.Dialog {
+	return model.Dialog{
+		CallbackId:     "somecallbackid",
+		Title:          "Sample Confirmation Dialog",
+		IconURL:        "http://www.mattermost.org/wp-content/uploads/2016/04/icon.png",
+		Elements:       nil,
+		SubmitLabel:    "Confirm",
+		NotifyOnCancel: true,
+		State:          "somestate",
+	}
 }
 
 func (p *Plugin) crash() {
-	go func() {
-		<-time.NewTimer(time.Second).C
-		y := 0
-		y = 1 / y
-	}()
+	<-time.NewTimer(time.Second).C
```

Comment:
Does the "crashed" message still appear without go routine?

Reply:
He method is still called in there own go routine. I just moved the routine outside of the method. See https://github.com/mattermost/mattermost-plugin-demo/pull/28/files#diff-ae7d00408bc3cf4a08823e84a6065b03R104

Reply:
Ah nice.  I wasn't familiar with that syntax and overlooked it.